integration-tests/camel-jms/with-named-and-external/src/main/java/io/quarkus/it/artemis/camel/jms/withnamedandexternal/BeanProducer.java

```diff
@@ -17,6 +18,7 @@ public class BeanProducer {
     @Typed({ ConnectionFactory.class })
     @ApplicationScoped
     @Identifier("externally-defined")
+    @Named("externally-defined")
```

Comment:
I don't like the fact that the externally defined connection factory needs the additional `@Named(...)` annotation. One goal of this feature for me is/was that `@Named(...)` is not necessary, the `CamelContextEnhancer` took care of this. 

From the point of view of users, it is hard to understand why they should add two annotations to the bean, definen the same name. Do we have other possibilities to - on a global level - tell camel that all `@Identifier(...)` beans should be part of the camel context? Is there a specific reason why we should remove the `CamelContextEnhancer`?

Reply:
Hm... I'll test the water with `camel-quarkus` on this feature. Do you see any harm in keeping the `CamelContextEnhancer` until then?

Reply:
No, if we decide to keep `CamelContextEnhancer` until we find a better way to do with `@Identifier` beans, I think it should be good to introduce a config property such like `quarkus.artemis.camel-quarkus-enhance-enabled` with default value `false` to protect this feature. And explain it in the document, if users define a external ConnectionFactory bean with `@Identifier`, it should enable this propery to make it availiable in the camel router definition unless add `@Named`. WDYT?

Reply:
Sounds good to me. Should we close this PR, and I'll take care of enabling the context enhancer through a property?

Reply:
Yeah, I think so and please keep in mind to add `configurator.name(name);` in `ArtemisJmsProcessor`. Thanks!